db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OptimizerImpl.java

```diff
@@ -2790,4 +2794,25 @@ public int getJoinPosition() {
     public CostModel getCostModel() {
         return costModel;
     }
+
+    @Override
+    public void setOuterTableOfJoin(ResultSetNode outerTableOfJoin) {
+        if (outerTableOfJoin == null)
+            this.outerTableOfJoin = null;
+        if (outerTableOfJoin instanceof Optimizable)
+            this.outerTableOfJoin = (Optimizable)outerTableOfJoin;
```

Comment:
If outerTableOfJoin is not null and not instanceof Optimizable, is it ok or should be an Exception?

Reply:
This is a good question.  I have modified setOuterTableOfJoin to save the outer table, even if it is not an Optimizable.  A trigger VTI will always be an Optimizable.  But this general-purpose function should always record the outer table, even if it is not an Optimizable.